packages/popover/src/positionUtils.ts

```diff
@@ -122,8 +146,21 @@ export function getElementDocumentPosition(
     return defaultElementPosition;
   }
 
-  const { top, bottom, left, right } = element.getBoundingClientRect();
-  const { offsetHeight: height, offsetWidth: width } = element;
+  const {
+    top,
+    bottom,
+    left,
+    right,
+    width: boundingWidth,
+  } = element.getBoundingClientRect();
+  const { offsetHeight: height, offsetWidth } = element;
+
+  // offsetWidth returns a rounded number of the element's layout width and height.
+  // boundingWidth returns an exact number with the rendered width and height which can include transformations.
+  // Using the exact number is a better indicator of determining if an element will fit in the window, e.g. calcLeft() uses the width to determine the left position and then that number is used to check if it is safetly within the window bounds. In some cases the offsetWidth is rounded up which means that it might not fit in the window bounds when it really can.
+  // With this we check if the difference between the boundingWidth and offsetWidth is less than one, if thats the case then the number was rounded to a whole number and we should use the exact number instead. However if the number is greater than one then that means that the boundingWidth is returning a width that has a transformation applied to it and we don't want that number. We want the untransformed width.
+  const width =
+    Math.abs(boundingWidth - offsetWidth) < 1 ? boundingWidth : offsetWidth;
```

Comment:
🐞 Small nit: extracting `Math.abs(boundingWidth - offsetWidth) < 1` into a var called something like `wasRounded` would make this a bit clearer